playbooks/roles/mattermost/templates/config.json.j2

```diff
@@ -127,32 +186,37 @@
         "EnableSignInWithEmail": true,
         "EnableSignInWithUsername": true,
         "SendEmailNotifications": true,
-        "RequireEmailVerification": false,
-        "FeedbackName": "Mattermost Notification",
+        "RequireEmailVerification": {{ MATTERMOST_REQUIRE_EMAIL_VERIFICATION }},
+        "FeedbackName": "{{ MATTERMOST_FEEDBACK_NAME }}",
         "FeedbackEmail": "{{ MATTERMOST_FEEDBACK_EMAIL }}",
         "FeedbackOrganization": "",
-        "EnableSMTPAuth": false,
-        "SMTPUsername": "",
-        "SMTPPassword": "",
-        "SMTPServer": "localhost",
-        "SMTPPort": "25",
-        "ConnectionSecurity": "",
+        "SMTPUsername": "{{ MATTERMOST_SMTP_USERNAME }}",
+        "SMTPPassword": "{{ MATTERMOST_SMTP_PASSWORD }}",
+        "SMTPServer": "{{ MATTERMOST_SMTP_SERVER }}",
+        "SMTPPort": "{{ MATTERMOST_SMTP_PORT }}",
+        "ConnectionSecurity": "TLS",
         "InviteSalt": "{{ MATTERMOST_INVITE_SALT }}",
         "SendPushNotifications": true,
-        "PushNotificationServer": "",
+        "PushNotificationServer": "{{ MATTERMOST_PUSH_NOTIFICATION_SERVER }}",
         "PushNotificationContents": "generic",
-        "EnableEmailBatching": false,
+        "EnableEmailBatching": true,
         "EmailBatchingBufferSize": 256,
         "EmailBatchingInterval": 30,
         "SkipServerCertificateVerification": false
+        {% if {{ MATTERMOST_VERSION }} != '4.0.0' %}
+            "SkipServerCertificateVerification": false,
```

Comment:
It doesn't make sense to test on MATTERMOST_VERSION unless what you are testing for is version dependent. Since I don't see any tests based on host I am guess you are using version as a stand in for host? It would be better to use inventory host directly. 

This `SkipServerCertificateVerification` line occurs inside and outside the block.

Reply:
I'm not entirely sure on what basis the config differs i.e. whether it depends only on version or something else. I figured testing based on version would make more sense since we use an older version and so it's possible some of the features are not supported. Also none of the other services using the same playbook are using it (default version was set to 5). 
Are you suggesting testing against particular hosts? i.e. we'll have a check like `if {{ MATTERMOST_HOSTNAME }} == 'x.x.x.x'`. That value will need to be changed for each host right? Or is there another way to do that?